syntax = "proto3";

package minetorch;

service Minetorch {
  rpc HeyYo (HeyMessage) returns (YoMessage) {}
  rpc SendLog (Log) returns (StandardResponse) {}
  rpc SetTimer (Timer) returns (StandardResponse) {}
  rpc AddPoint (Point) returns (StandardResponse) {}
}

message StandardResponse {
  int32 status = 1;
  string message = 2;
}

message HeyMessage {
  enum Status {
    IDLE = 0;
    TRAINING = 1;
    ERROR = 2;
  };

  string ip_addr = 1;
  int32 experiment_id = 2;
  string experiment_name = 3;
  Status status = 4;
}

message YoMessage {
  enum Command {
    TRAIN = 0;
    HALT = 1;
    KILL = 2;
  };

  int32 experiment_id = 1;
  bool roger = 2;
  Command command = 3;
}

message Log {
  int32 experiment_id = 1;
  string log = 2;
  string level = 3;
}

message Timer {
  enum Category {
    ITERATION = 0;
    EPOCH = 1;
    SNAPSHOT = 2;
  };

  int32 experiment_id = 1;
  string name = 2;
  Category category = 3;
  int32 current = 4;
}

message Point {
  int32 experiment_id = 1;
  string graph_name = 2;
  int32 x = 3;
  float y = 4;
}
